apps/services/user-notification/src/app/modules/notifications/notificationsWorker/notificationsWorker.service.ts

```diff
@@ -310,155 +305,310 @@ export class NotificationsWorkerService {
+        false,
+        { nationalId: message.recipient } as User,
+      )
+
+      if (allowCompanyUserProfileEmails) {
+        userProfile =
+          await this.userProfileApi.userProfileControllerFindUserProfile({
+            xParamNationalId: message.recipient,
+          })
+
+        if (userProfile) {
+          locale = (userProfile.locale as Locale) || 'is'
         }
+      }
+    }
 
-        await Promise.all(notificationPromises)
-      },
+    const template = await this.notificationsService.getTemplate(
+      message.templateId,
+      locale,
+    )
+
+    const scope = template.scope || DocumentsScope.main
+    const notification = await this.createUserNotificationDbRecord(args, scope)
+
+    const shouldSendNotifications =
+      (!isCompanyRecipient && userProfile) ||
+      (isCompanyRecipient && userProfile && allowCompanyUserProfileEmails)
+
+    if (shouldSendNotifications && userProfile) {
+      const handleNotificationArgs: HandleNotification = {
+        profile: userProfile,
+        messageId: args.messageId,
+        notificationId: notification?.id,
+        message: args,
+        template,
+      }
+      await this.handleEmailNotification(handleNotificationArgs)
+      await this.handlePushNotifications(handleNotificationArgs)
+    }
+
+    await this.handleSendingNotificationsToDelegations(
+      args,
+      scope,
+      actorNationalId,
     )
   }
 
+  private async handleSendingNotificationsToDelegations(
+    args: CreateHnippNotificationDto & { messageId: string },
+    templateScope: string,
+    actorNationalId?: string,
+  ) {
+    const { messageId, ...message } = args
+
+    // Only proceed if the template scope is in the allowed notification scopes
+    if (!notificationScopes.includes(templateScope)) {
+      this.logger.info('Template scope is not in allowed notification scopes', {
+        templateScope,
+        messageId,
+      })
+      return
+    }
+
+    try {
+      const delegations =
+        await this.delegationsApi.delegationsControllerGetDelegationRecords({
+          xQueryNationalId: message.recipient,
+          scopes: templateScope,
+          direction:
+            DelegationsControllerGetDelegationRecordsDirectionEnum.outgoing,
+        })
+
+      let recipientName = ''
+
+      if (delegations.data.length > 0) {
+        recipientName = await this.getName(message.recipient)
+      }
+
+      // Filter out duplicate delegations that have the same fromNationalId and toNationalId
+      delegations.data = delegations.data.filter(
+        (delegation, index, self) =>
+          index ===
+          self.findIndex(
+            (d) =>
+              d.fromNationalId === delegation.fromNationalId &&
+              d.toNationalId === delegation.toNationalId,
+          ),
+      )
+
+      const delegationsToSend = actorNationalId
+        ? delegations.data.filter(
+            (delegation) => actorNationalId === delegation.toNationalId,
+          )
```

Comment:
Do we want to fallback to the other delegations if we can't find the requested actor (i.e. they don't have a delegation anymore)?